src/p_spec.c

```diff
@@ -2638,6 +2641,44 @@ void T_Scroll(scroll_t *s)
     }
 }
 
+// [crispy] smooth texture scrolling
+
+static int numsidescrollers;
+static scroll_t **sidescrollers;
+
+static void P_AddSideScroller (scroll_t *s)
+{
+  if (s)
+  {
+    sidescrollers = I_Realloc(sidescrollers, (numsidescrollers + 1) * sizeof(*sidescrollers));
```

Comment:
Theoretically, there can be many sidescrollers on the level. Maybe it makes sense to allocate less often?